File: 프로그래머스/unrated/181833. 특별한 이차원 배열 1/특별한 이차원 배열 1.js
function solution(n) {
    const arr = [];
    
    for (let i = 0; i < n; i++) {
        const arr2 = new Array(n).fill(0);
        arr.push(arr2);
    }
    
    for (let i = 0; i < n; i++) {
        arr[i][i] = 1;
    }
    
    return arr;
}

File: 프로그래머스/unrated/181833. 특별한 이차원 배열 1/특별한 이차원 배열 1.java
class Solution {
    public int[][] solution(int n) {
        int[][] answer = new int[n][n];
        
        for (int i = 0; i < n; i++) {
            int[] temp = new int[n];
            for (int j = 0; j < n; j++) {
                if (i == j) {
                    temp[j] = 1;
                } else {
                    temp[j] = 0;
                }
            }
            answer[i] = temp;
        }
        
        return answer;
    }
}